File: guides/java/kora-java-guide-dependency-injection-introduction-app/src/main/java/ru/tinkoff/kora/guide/dependencyinjection/SmsNotifier.java
package ru.tinkoff.kora.guide.dependencyinjection;

import ru.tinkoff.kora.common.Component;
import ru.tinkoff.kora.common.Tag;

@Tag(SmsTag.class)
@Component
public final class SmsNotifier implements Notifier {

    private final MessageFormatter formatter;

    public SmsNotifier(MessageFormatter formatter) {
        this.formatter = formatter;
    }

    @Override
    public String channel() {
        return "sms";
    }

    @Override
    public String notifyUser(String message) {
        return "SMS: " + formatter.format(message);
    }
}

File: guides/kotlin/kora-kotlin-guide-dependency-injection-introduction-app/src/main/kotlin/ru/tinkoff/kora/guide/dependencyinjection/SmsNotifier.kt
package ru.tinkoff.kora.guide.dependencyinjection

import ru.tinkoff.kora.common.Component
import ru.tinkoff.kora.common.Tag

@Tag(SmsTag::class)
@Component
class SmsNotifier(
    private val formatter: MessageFormatter
) : Notifier {

    override fun channel(): String = "sms"

    override fun notifyUser(message: String): String = "SMS: ${formatter.format(message)}"
}
